### ISSUE TITLE
Error - Config is invalid

### ISSUE THREAD

**Issue by akashnimare** [electron-userland/electron-builder#1418]
* **Version**: 
14.1.1 && 16.4.2

* **Target**: 
macOS

`npm run dist`
output - 
```sh

> zulip@0.5.8 dist /Users/akka/dev/zulipwork/zulip-electron
> build

Error: Config is invalid:
{
  "asar": [
    "Should be object",
    "Should be null,boolean",
    "Invalid option object"
  ],
  "linux": [
    {
      "synopsis": "Unknown option",
      "version": "Unknown option",
      "title": "Unknown option",
      "license": "Unknown option"
    },
    "Invalid option object"
  ],
  "win": [
    {
      "iconUrl": "Unknown option",
      "target": [
        "Should be array",
        "Invalid option object"
      ]
    },
    "Invalid option object"
  ]
}

Raw validation errors: [
  {
    "keyword": "type",
    "dataPath": ".asar",
    "schemaPath": "#/definitions/AsarOptions/type",
    "params": {
      "type": "object"
    },
    "message": "should be object"
  },
  {
    "keyword": "type",
    "dataPath": ".asar",
    "schemaPath": "#/properties/asar/anyOf/1/type",
    "params": {
      "type": "null,boolean"
    },
    "message": "should be null,boolean"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".asar",
    "schemaPath": "#/properties/asar/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  },
  {
    "keyword": "additionalProperties",
    "dataPath": ".linux",
    "schemaPath": "#/additionalProperties",
    "params": {
      "additionalProperty": "synopsis"
    },
    "message": "should NOT have additional properties"
  },
  {
    "keyword": "additionalProperties",
    "dataPath": ".linux",
    "schemaPath": "#/additionalProperties",
    "params": {
      "additionalProperty": "version"
    },
    "message": "should NOT have additional properties"
  },
  {
    "keyword": "additionalProperties",
    "dataPath": ".linux",
    "schemaPath": "#/additionalProperties",
    "params": {
      "additionalProperty": "title"
    },
    "message": "should NOT have additional properties"
  },
  {
    "keyword": "additionalProperties",
    "dataPath": ".linux",
    "schemaPath": "#/additionalProperties",
    "params": {
      "additionalProperty": "license"
    },
    "message": "should NOT have additional properties"
  },
  {
    "keyword": "type",
    "dataPath": ".linux",
    "schemaPath": "#/properties/linux/anyOf/1/type",
    "params": {
      "type": "null"
    },
    "message": "should be null"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".linux",
    "schemaPath": "#/properties/linux/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  },
  {
    "keyword": "additionalProperties",
    "dataPath": ".win",
    "schemaPath": "#/additionalProperties",
    "params": {
      "additionalProperty": "iconUrl"
    },
    "message": "should NOT have additional properties"
  },
  {
    "keyword": "type",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf/0/type",
    "params": {
      "type": "array"
    },
    "message": "should be array"
  },
  {
    "keyword": "type",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf/1/type",
    "params": {
      "type": "null"
    },
    "message": "should be null"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  },
  {
    "keyword": "type",
    "dataPath": ".win",
    "schemaPath": "#/properties/win/anyOf/1/type",
    "params": {
      "type": "null"
    },
    "message": "should be null"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".win",
    "schemaPath": "#/properties/win/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  }
]
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/src/packager.ts:139:13
    at next (native)
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/jsonfile/index.js:46:5
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:446:3)
From previous event:
    at Packager.build (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/packager.js:278:11)
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/src/builder.ts:217:33
    at next (native)
From previous event:
    at build (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/builder.js:70:21)
    at Object.<anonymous> (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/cli/build-cli.js:71:41)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:393:7)
    at startup (bootstrap_node.js:150:9)
    at bootstrap_node.js:508:3

npm ERR! Darwin 16.4.0
npm ERR! argv "/Users/akka/.nvm/versions/node/v6.10.1/bin/node" "/Users/akka/.nvm/versions/node/v6.10.1/bin/npm" "run" "dist"
npm ERR! node v6.10.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! zulip@0.5.8 dist: `build`
npm ERR! Exit status 255
npm ERR! 
npm ERR! Failed at the zulip@0.5.8 dist script 'build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the zulip package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs zulip
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls zulip
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/akka/dev/zulipwork/zulip-electron/npm-debug.log


```
Package.json - https://github.com/zulip/zulip-electron/blob/master/package.json
App's package.json - https://github.com/zulip/zulip-electron/blob/master/app/package.json

**Comment by develar** [comment]
> "asar": "true",

Must be `"asar": true,`

**Comment by develar** [comment]
>  "iconUrl": "https://raw.githubusercontent.com/zulip/zulip-electron/master/build/icon.ico"

Not required anymore.

**Comment by akashnimare** [comment]
@develar changed it [here](https://github.com/zulip/zulip-electron/commit/3a6bb142248b96e583bc6d3faa7bcebc219fc305), still getting the same error. Is there anything which needs to be changed?

**Comment by develar** [comment]
@akashnimare 

```
diff --git a/package.json b/package.json
index a9c55b8..ba4ae70 100644
--- a/package.json
+++ b/package.json
@@ -40,7 +40,6 @@
       "category": "public.app-category.productivity"
     },
     "linux": {
-      "synopsis": "Zulip Desktop App",
       "category": "",
       "packageCategory": "GNOME;GTK;Network;InstantMessaging",
       "description": "Zulip Desktop Client for Linux",
@@ -48,11 +47,11 @@
         "deb",
         "AppImage"
       ],
-      "version": "0.5.8",
-      "title": "Zulip",
-      "license": "Apache-2.0",
       "maintainer": "Akash Nimare <svnitakash@gmail.com>"
     },
+    "deb": {
+      "synopsis": "Zulip Desktop App"
:

```

**Comment by akashnimare** [comment]
Still no luck... updated commit - https://github.com/zulip/zulip-electron/commit/0ad66399d0309b9da34e66a7014ef55cfbb85072

**Comment by akashnimare** [comment]
Updated error log -
```sh

> zulip@0.5.8 dist /Users/akka/dev/zulipwork/zulip-electron
> build

Error: Config is invalid:
{
  "win": [
    {
      "target": [
        "Should be array",
        "Invalid option object"
      ]
    },
    "Invalid option object"
  ]
}

Raw validation errors: [
  {
    "keyword": "type",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf/0/type",
    "params": {
      "type": "array"
    },
    "message": "should be array"
  },
  {
    "keyword": "type",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf/1/type",
    "params": {
      "type": "null"
    },
    "message": "should be null"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".win.target",
    "schemaPath": "#/properties/target/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  },
  {
    "keyword": "type",
    "dataPath": ".win",
    "schemaPath": "#/properties/win/anyOf/1/type",
    "params": {
      "type": "null"
    },
    "message": "should be null"
  },
  {
    "keyword": "anyOf",
    "dataPath": ".win",
    "schemaPath": "#/properties/win/anyOf",
    "params": {},
    "message": "should match some schema in anyOf"
  }
]
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/src/packager.ts:139:13
    at next (native)
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/jsonfile/index.js:46:5
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:445:3)
From previous event:
    at Packager.build (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/packager.js:278:11)
    at /Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/src/builder.ts:217:33
    at next (native)
From previous event:
    at build (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/builder.js:70:21)
    at Object.<anonymous> (/Users/akka/dev/zulipwork/zulip-electron/node_modules/electron-builder/out/cli/build-cli.js:71:41)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3

npm ERR! Darwin 16.4.0
npm ERR! argv "/Users/akka/.nvm/versions/node/v6.9.4/bin/node" "/Users/akka/.nvm/versions/node/v6.9.4/bin/npm" "run" "dist"
npm ERR! node v6.9.4
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! zulip@0.5.8 dist: `build`
npm ERR! Exit status 255
npm ERR! 
npm ERR! Failed at the zulip@0.5.8 dist script 'build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the zulip package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs zulip
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls zulip
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/akka/dev/zulipwork/zulip-electron/npm-debug.log

```

**Comment by develar** [comment]
> "electron-builder": "14.1.1",

Please latest version.

**Comment by akashnimare** [comment]
@develar fixed via v16.6.0. Thanks a lot for the help ❤️
